client/blocks/keyring-connect-button/index.js

```diff
@@ -0,0 +1,239 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { last, isEqual, some } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import {
+	deleteStoredKeyringConnection,
+	requestKeyringConnections,
+} from 'state/sharing/keyring/actions';
+import { getKeyringServiceByName } from 'state/sharing/services/selectors';
+import QueryKeyringServices from 'components/data/query-keyring-services';
+import requestExternalAccess from 'lib/sharing';
+import { getKeyringConnectionsByName } from 'state/sharing/keyring/selectors';
+import { isFetchingConnections } from 'state/sharing/publicize/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+class KeyringConnectButton extends Component {
+	static propTypes = {
+		service: PropTypes.oneOfType( [ PropTypes.object, PropTypes.bool ] ),
```

Comment:
`service` is used as an object, so IMO it should be an `object` here, of at least shape:
```js
PropTypes.shape( {
    ID: PropTypes.string.isRequired,
    connect_URL: PropTypes.string.isRequired,
} )
```

Reply:
`getKeyringServiceByName` may return `false` if keyring services has not loaded yet or no service is found
https://github.com/Automattic/wp-calypso/blob/master/client/state/sharing/services/selectors.js#L44